Project cannot be built

The project uses TypeScript to build

https://github.com/holocron-hq/safe-mdx/blob/66aa3dbf12d4254dfa1b622ff7524ddb3928339c/package.json#L19

but TypeScript is not included in the dependencies

https://github.com/holocron-hq/safe-mdx/blob/66aa3dbf12d4254dfa1b622ff7524ddb3928339c/package.json#L35-L52

TypeScript should be included as a `devDependency` to track the version used to build.

In addition, the TypeScript configuration file is incomplete, it references a relative file outside the repository

https://github.com/holocron-hq/safe-mdx/blob/66aa3dbf12d4254dfa1b622ff7524ddb3928339c/tsconfig.json#L2

When building with TypeScript 5 this results in the error:
```
error TS5083: Cannot read file '../tsconfig.base.json'.
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -1,0 +1,21 @@
+name: main
+on:
+  - pull_request
+  - push
+jobs:
+  main:
+    name: ${{matrix.node}}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{matrix.node}}
+      - run: npm install
+      - run: npm run build
+      - run: npm test
+    strategy:
+      matrix:
+        node:
+          - lts/gallium
+          - node

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
 node_modules
 .DS_Store
 dist
+coverage

--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
     },
     "scripts": {
         "build": "tsc",
-        "test": "vitest",
+        "test": "vitest run --coverage",
         "prepublishOnly": "pnpm build"
     },
     "files": [
@@ -33,21 +33,23 @@
         "react": "*"
     },
     "dependencies": {
-        "html-to-jsx-transform": "^1.1.0",
-        "mdast": "^3.0.0",
-        "remark": "^15.0.1",
+        "html-to-jsx-transform": "^1.0.0",
+        "remark": "^15.0.0",
         "remark-frontmatter": "^5.0.0",
         "remark-gfm": "^4.0.0",
         "remark-mdx": "^3.0.0"
     },
     "devDependencies": {
-        "@types/mdast": "^4.0.3",
-        "@types/node": "^20.11.13",
-        "@types/react": "18.2.48",
-        "@types/react-dom": "^18.2.18",
-        "dedent": "^1.2.0",
+        "@types/mdast": "^4.0.0",
+        "@types/node": "^20.0.0",
+        "@types/react": "^18.0.0",
+        "@types/react-dom": "^18.0.0",
+        "@vitest/coverage-v8": "^1.0.0",
+        "dedent": "^1.0.0",
         "mdast-util-mdx-jsx": "^3.0.0",
-        "react": "^18.2.0",
-        "react-dom": "^18.2.0"
+        "react": "^18.0.0",
+        "react-dom": "^18.0.0",
+        "typescript": "^5.0.0",
+        "vitest": "^1.0.0"
     }
 }

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,1 +1,1 @@
-export * from './safe-mdx'
+export * from './safe-mdx.js'

--- a/src/safe-mdx.test.tsx
+++ b/src/safe-mdx.test.tsx
@@ -1,7 +1,7 @@
 import React from 'react'
 import dedent from 'dedent'
 import { test, expect } from 'vitest'
-import { MdastToJsx } from './safe-mdx'
+import { MdastToJsx } from './safe-mdx.js'
 import { renderToStaticMarkup } from 'react-dom/server'
 void React
 

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,10 +1,13 @@
 {
-    "extends": "../tsconfig.base.json",
     "compilerOptions": {
-        "rootDir": "src",
+        "target": "es2022",
         "jsx": "react-jsx",
-        "module": "ES2015",
-        "outDir": "dist"
+        "module": "node16",
+        "outDir": "dist",
+        "strict": false,
+        "declaration": true,
+        "sourceMap": true,
+        "declarationMap": true,
     },
     "include": ["src"]
 }

